<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>An R package utilized to refine the ACMG/AMP criteria according to the Bayesian framework • BayesQuantify</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="An R package utilized to refine the ACMG/AMP criteria according to the Bayesian framework">
<meta name="description" content="The guidelines proposed by the American College of Medical Genetics and Genomics (ACMG) and the Association for Molecular Pathology (AMP)have undergone continuous review and refinement for different rules, genes, and diseases, driving optimization and enhancing variant interpretation standards in genetic testing. In 2018, the Clinical Genome Resource (ClinGen) Sequence Variant Interpretation (SVI) Working Group has proposed a Bayesian Classification Framework to model the ACMG/AMP guidelines. This framework has successfully quantified the thresholds for applying PM5 and PP3/BP4 criteria. However, existing software and tools designed for quantifying the evidence strength and establishing corresponding thresholds to refine the ACMG/AMP criteria are lacking. This package provide users with a unified resource for quantifying the strength of evidence for ACMG/AMP criteria using a naive Bayes classifier.">
<meta property="og:description" content="The guidelines proposed by the American College of Medical Genetics and Genomics (ACMG) and the Association for Molecular Pathology (AMP)have undergone continuous review and refinement for different rules, genes, and diseases, driving optimization and enhancing variant interpretation standards in genetic testing. In 2018, the Clinical Genome Resource (ClinGen) Sequence Variant Interpretation (SVI) Working Group has proposed a Bayesian Classification Framework to model the ACMG/AMP guidelines. This framework has successfully quantified the thresholds for applying PM5 and PP3/BP4 criteria. However, existing software and tools designed for quantifying the evidence strength and establishing corresponding thresholds to refine the ACMG/AMP criteria are lacking. This package provide users with a unified resource for quantifying the strength of evidence for ACMG/AMP criteria using a naive Bayes classifier.">
<meta property="og:image" content="/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">BayesQuantify</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/BayesQuantify.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="bayesquantify-">BayesQuantify <a class="anchor" aria-label="anchor" href="#bayesquantify-"></a>
</h1>
</div>
<!-- badges: start -->

<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The ACMG/AMP guidelines have undergone continuous review and refinement for different rules, genes, and diseases, driving optimization and enhancing variant interpretation standards in genetic testing. In 2018, the ClinGen Sequence Variant Interpretation Working Group has proposed a Bayesian Classification Framework to model the ACMG/AMP guidelines. This framework has successfully quantified the thresholds for applying PM5 and PP3/BP4. However, existing software and tools designed for quantifying the evidence strength and establishing corresponding thresholds to refine the ACMG/AMP criteria are lacking.</p>
<p>The <code>BayesQuantify</code> R Package provides users with a unified resource for quantifying the strength of evidence for ACMG/AMP criteria using a naive Bayes classifier. The functions included in BayesQuantify consist of five main steps (as shown in Figure 1):</p>
<div class="float">
<img src="https://github.com/liusihan/BayesQuantify/blob/master/man/figures/Figure1_BayesQuantify.png?raw=true" alt="Figure 1"><div class="figcaption">Figure 1</div>
</div>
<p align="center">
Figure 1. Schematic overview of BayesQuantify.
</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <code>BayesQuantify</code> from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"liusihan/BayesQuantify"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="required-input-data">Required Input Data<a class="anchor" aria-label="anchor" href="#required-input-data"></a>
</h2>
<p>The package requires one essential input: a data.frame containing variants classification results, including the variants, ACMG/AMP evidence, variant classification, and tested features.</p>
<p>Two distinct datasets: the ClinGen Curated Variants dataset and the ClinVar 2019 dataset have been included in <code>BayesQuantify</code>. Here is an example input file：</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="16%">
<col width="16%">
<col width="16%">
<col width="16%">
<col width="16%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th>#Variation</th>
<th>Assertion</th>
<th>Applied Evidence Codes (Met)</th>
<th>HGNC Gene Symbol</th>
<th>Expert Panel</th>
<th>…</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>NM_000277.2:c.1A&gt;G</td>
<td>Pathogenic</td>
<td>PS3, PM3, PP4_Moderate, PM2</td>
<td>PAH</td>
<td>Phenylketonuria VCEP</td>
<td>…</td>
</tr>
<tr class="even">
<td>NM_000314.6:c.209+3A&gt;T</td>
<td>Uncertain Significance</td>
<td>PP3</td>
<td>PTEN</td>
<td>PTEN VCEP</td>
<td>…</td>
</tr>
<tr class="odd">
<td>NM_004999.3:c.2836C&gt;T</td>
<td>Likely Benign</td>
<td>BS1</td>
<td>MYO6</td>
<td>Hearing Loss VCEP</td>
<td>…</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>The full usage of BayesQuantify can be found <a href="https://github.com/liusihan/BayesQuantify/blob/207dfa884c0ac84b8ed281ece95dee5e1b3a8fb6/man/figures/BayesQuantify.pdf" class="external-link">here</a>.</p>
<p>You can find a step-by-step example of how to use it below:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">library</span>(BayesQuantify)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="do">## Step 1: Calculation of OP and Post_P for each evidence strength with user-defined Prior_P.</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># Prior_P=0.1 was recommended by ClinGen SVI Working Group</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">auto_select_postp</span>(<span class="fl">0.1</span>)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  Evidence Strength Odds of pathogenicity Posterior probability of pathogenicity and benignity</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="dv">1</span>               PVS                   <span class="dv">351</span>                                                <span class="fl">0.975</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="dv">2</span>                PS      <span class="fl">18.7349939951952</span>                                     <span class="fl">0.67550020016016</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="dv">3</span>                PM      <span class="fl">4.32839392791312</span>                                    <span class="fl">0.324749849931156</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="dv">4</span>                PP      <span class="fl">2.08047925438182</span>                                    <span class="fl">0.187760764369383</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="dv">5</span>                BP     <span class="fl">0.480658481882885</span>                                    <span class="fl">0.949301150041276</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="dv">6</span>                BM     <span class="fl">0.231032576205959</span>                                    <span class="fl">0.974972184931784</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="dv">7</span>                BS    <span class="fl">0.0533760512683624</span>                                    <span class="fl">0.994104293142569</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="dv">8</span>               BVS   <span class="fl">0.00284900284900285</span>                                    <span class="fl">0.999683544303797</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="do">## Step 2: Variants of US (VUS) subclassification</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co"># This step will categorize VUS into six confidence tiers (hot, warm, tepid, cool, cold, and ice cold) according to</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co"># the Association for Clinical Genomic Science (ACGS) Best Practice Guidelines.</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">data</span>(<span class="st">"ClinGen_dataset"</span>)</span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="sc">&gt;</span> data <span class="ot">&lt;-</span> <span class="fu">add_info</span>(ClinGen_dataset, <span class="st">"Assertion"</span>)</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="sc">&gt;</span> data <span class="ot">&lt;-</span> <span class="fu">VUS_classify</span>(data, <span class="st">"Assertion"</span>, <span class="st">"Applied Evidence Codes (Met)"</span>)</span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="do">## Step3: Calculating LR or lr</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co"># To ensure a robust and reliable truth set for evidence calibration, we recommend excluding hot, warm, tepid VUS.</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co"># For binary variables: </span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="sc">&gt;</span> data<span class="ot">&lt;-</span><span class="fu">discrete_cutoff</span>(data, <span class="st">"Applied Evidence Codes (Met)"</span>, <span class="at">criteria =</span> <span class="st">"PM2_Supporting"</span>)</span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="sc">&gt;</span> data<span class="ot">&lt;-</span><span class="fu">discrete_cutoff</span>(data, <span class="st">"Applied Evidence Codes (Met)"</span>, <span class="at">criteria =</span> <span class="st">"PM2"</span>)</span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">library</span>(dplyr)</span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="sc">&gt;</span> truth_set <span class="ot">&lt;-</span> <span class="fu">filter</span>(data,VUS_class <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"IceCold"</span>,<span class="st">"Cold"</span>,<span class="st">"Cool"</span>,<span class="st">""</span>))</span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="sc">&gt;</span> LR_result<span class="ot">&lt;-</span><span class="fu">LR</span>(truth_set, <span class="dv">28</span>, <span class="dv">29</span>)</span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a></span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a><span class="co"># For quantitative variables: </span></span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">data</span>(<span class="st">"ClinVar_2019_dataset"</span>)</span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a><span class="sc">&gt;</span> data <span class="ot">&lt;-</span> <span class="fu">add_info</span>(ClinVar_2019_dataset, <span class="st">"clnsig"</span>)</span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">local_bootstrapped_lr</span>(data, <span class="st">"PrimateAI_score"</span>, <span class="st">"Pathogenic"</span>,<span class="fl">0.0441</span>, <span class="dv">10000</span>, <span class="dv">100</span>, <span class="fl">0.01</span>, <span class="st">"test_dir"</span>)</span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a><span class="sc">&gt;</span> postp_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.100</span>, <span class="fl">0.211</span>, <span class="fl">0.608</span>, <span class="fl">0.981</span>)</span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a><span class="sc">&gt;</span> lr_CI_result <span class="ot">&lt;-</span> <span class="fu">lr_CI</span>(<span class="dv">10000</span>, <span class="st">"test_dir"</span>)</span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">get_lr_threshold</span>(lr_CI_result, postp_list, <span class="st">"Pathogenic"</span>)</span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a></span>
<span id="cb2-40"><a href="#cb2-40" tabindex="-1"></a><span class="do">## Step 4: Variant classification</span></span>
<span id="cb2-41"><a href="#cb2-41" tabindex="-1"></a><span class="co"># BayesQuantify offers two functions for variant classification</span></span>
<span id="cb2-42"><a href="#cb2-42" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">data</span>(<span class="st">"ClinGen_dataset"</span>)</span>
<span id="cb2-43"><a href="#cb2-43" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">ACMG_Classification</span>(ClinGen_dataset, <span class="st">"Applied Evidence Codes (Met)"</span>)</span>
<span id="cb2-44"><a href="#cb2-44" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">BCF</span>(ClinGen_dataset, <span class="st">"Applied Evidence Codes (Met)"</span>, <span class="fl">0.1</span>, <span class="dv">350</span>)</span>
<span id="cb2-45"><a href="#cb2-45" tabindex="-1"></a></span>
<span id="cb2-46"><a href="#cb2-46" tabindex="-1"></a><span class="do">## Step 5: Visualization</span></span>
<span id="cb2-47"><a href="#cb2-47" tabindex="-1"></a><span class="co"># Display variant characteristics</span></span>
<span id="cb2-48"><a href="#cb2-48" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">data</span>(<span class="st">"ClinGen_dataset"</span>)</span>
<span id="cb2-49"><a href="#cb2-49" tabindex="-1"></a><span class="sc">&gt;</span> ClinGen_dataset <span class="ot">&lt;-</span> <span class="fu">add_info</span>(ClinGen_dataset, <span class="st">"Assertion"</span>)</span>
<span id="cb2-50"><a href="#cb2-50" tabindex="-1"></a><span class="sc">&gt;</span> ClinGen_dataset <span class="ot">&lt;-</span> <span class="fu">VUS_classify</span>(ClinGen_dataset, <span class="st">"Assertion"</span>, <span class="st">"Applied Evidence Codes (Met)"</span>)</span>
<span id="cb2-51"><a href="#cb2-51" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">multi_plot</span>(ClinGen_dataset, <span class="st">"Assertion"</span>, <span class="st">"HGNC Gene Symbol"</span>)</span>
<span id="cb2-52"><a href="#cb2-52" tabindex="-1"></a></span>
<span id="cb2-53"><a href="#cb2-53" tabindex="-1"></a><span class="co"># For LR</span></span>
<span id="cb2-54"><a href="#cb2-54" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">data</span>(<span class="st">"LR_result"</span>)</span>
<span id="cb2-55"><a href="#cb2-55" tabindex="-1"></a><span class="sc">&gt;</span> op_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">2.08</span>, <span class="fl">4.33</span>, <span class="fl">18.70</span>, <span class="dv">351</span>)</span>
<span id="cb2-56"><a href="#cb2-56" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">heatmap_LR</span>(LR_result, <span class="st">"Pathogenic"</span>, op_list)</span>
<span id="cb2-57"><a href="#cb2-57" tabindex="-1"></a></span>
<span id="cb2-58"><a href="#cb2-58" tabindex="-1"></a><span class="co"># For lr</span></span>
<span id="cb2-59"><a href="#cb2-59" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">data</span>(<span class="st">"lr_CI_result"</span>)</span>
<span id="cb2-60"><a href="#cb2-60" tabindex="-1"></a><span class="sc">&gt;</span> postp_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.100</span>, <span class="fl">0.211</span>, <span class="fl">0.608</span>, <span class="fl">0.981</span>)</span>
<span id="cb2-61"><a href="#cb2-61" tabindex="-1"></a><span class="sc">&gt;</span> <span class="fu">plot_lr</span>(lr_CI_result, <span class="st">"Pathogenic"</span>, postp_list)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use BayesQuantify, please cite our paper (thanks!): &gt; Sihan Liu, Xiaoshu Feng, Fengxiao Bu. Calibration and refinement of ACMG/AMP criteria for variant classification with BayesQuantify[J]. medRxiv 2024.09.08.24313284; doi: <a href="https://doi.org/10.1101/2024.09.08.24313284" class="external-link uri">https://doi.org/10.1101/2024.09.08.24313284</a>.</p>
</div>
<div class="section level2">
<h2 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h2>
<p>If you encounter a clear bug, please file an issue with a minimal reproducible example on GitHub. For questions and other discussion, please contact Sihan Liu (<a href="mailto:liusihan@wchscu.cn" class="email">liusihan@wchscu.cn</a>).</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing BayesQuantify</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Sihan Liu <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/liusihan/BayesQuantify/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/liusihan/BayesQuantify/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/liusihan/BayesQuantify?branch=master" class="external-link"><img src="https://codecov.io/gh/liusihan/BayesQuantify/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Sihan Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
